src/shared/clients/ec2Client.ts

```diff
@@ -30,8 +35,25 @@ export class Ec2Client {
             'NextToken',
             'Reservations'
         )
-        const instances = this.getInstancesFromReservations(collection)
+        const extractedInstances = this.getInstancesFromReservations(collection)
+        const instances = await this.updateInstancesDetail(extractedInstances)
+
+        return instances
+    }
+
+    /** Updates status and name in-place for displaying to humans. */
+    protected async updateInstancesDetail(
+        instances: AsyncCollection<EC2.Instance>
+    ): Promise<AsyncCollection<EC2.Instance>> {
         return instances
+            .map(async instance => {
+                return { ...instance, status: await this.getInstanceStatus(instance.InstanceId!) }
+            })
+            .map(instance => {
+                return instanceHasName(instance!)
+                    ? { ...instance, name: lookupTagKey(instance!.Tags!, 'Name') }
+                    : instance!
+            })
```

Comment:
Each map() call re-scans the entire collection. Can both fields be updated in the same map() callback?